backend/api/image_generation.py

```
@@ -0,0 +1,329 @@
+FREEPIK_API_KEY = settings.freepik_api_key
+MAX_CONCURRENT_IMAGES = 10  # Limit concurrent image generations for scenes
+FREEPIK_BASE_URL = "https://api.freepik.com"
+
+# Global semaphore for image generation rate limiting (scenes)
+_image_semaphore = asyncio.Semaphore(MAX_CONCURRENT_IMAGES)
+
+# Log configuration on module load
+debug_log(f"[FREEPIK] FreePik configuration loaded: API Key available = {bool(FREEPIK_API_KEY)}")
+
+
+async def generate_scene_image(scene_description: str, scene_title: str, scenario_id: int = 0, scene_id: Optional[int] = None) -> str:
```

Ruff (0.14.3)

Unused function argument: `scenario_id`

(ARG001)